src/platform_impl/linux/window.rs

```diff
@@ -314,6 +314,83 @@ impl Window {
     Ok(win)
   }
 
+  pub(crate) fn new_from_gtk_window<T>(
+    event_loop_window_target: &EventLoopWindowTarget<T>,
+    window: gtk::ApplicationWindow,
+  ) -> Self {
```

Comment:
this should return a result, even though the logic deoesn;t have anything that propagates errors, we still should return result to make it consistent with `Window::new` and avoids breaking change in the futuer.

Reply:
Done.